src/rustdoc_gen.rs

```diff
@@ -110,6 +110,65 @@ impl<'a> CrateSource<'a> {
         all_crate_features.extend(self.regular_features());
         all_crate_features.into_iter().collect()
     }
+
+    /// Some feature names usually mean, that the feature is breaking semver by design.
+    /// This function filters them out.
```

Comment:
Probably a good idea to reference the following cargo issues here:
- unstable/nightly-only features: https://github.com/rust-lang/cargo/issues/10881
- private/hidden features: https://github.com/rust-lang/cargo/issues/10882
and explain how this heuristic relates to those. Most users probably haven't needed or used features that don't adhere to semver, so we should explain a bit of the context here so they understand what's going on.

For example:
```
Sometimes crates include features that are not meant for public use or otherwise don't adhere to semver:
- private features, like `bench` used for internal benchmarking only
- nightly-only features, like `nightly`
- unstable features containing experimental code, like `unstable`

To ensure the best possible out-of-the-box user experience, this function attempts to heuristically exclude feature names like the above.

This function uses heuristics based on the name since cargo does not currently include a mechanism for marking features as private/hidden/unstable. When such mechanisms are available in cargo, we'll update this functionality to make use of them. Relevant cargo issues:
- unstable/nightly-only features: https://github.com/rust-lang/cargo/issues/10881
- private/hidden features: https://github.com/rust-lang/cargo/issues/10882
```

Then include the description of the heuristic used etc.